[Pull request created only for viewing diff] Network interface syntax support starting from String to URI refactoring

irina-mitrea-luxoft opened the pull request:
**[Not to be merged]**

This pull request was created for facilitating code review and its sole purpose is to allow viewing the diff between "String to URI refactoring" story (branch [string_instead_of_uri](https://github.com/irina-mitrea-luxoft/gateway/tree/string_instead_of_uri)) and "Addition of network interface syntax" story (branch [network_interface_support](https://github.com/irina-mitrea-luxoft/gateway/tree/network_interface_support)) and **must not be merged**.

A separate pull request will be made with the diff between kaazing/gateway develop branch and irina-mitrea-luxoft/gateway network_interface_support branch.

irina-mitrea-luxoft commented:
@jitsni Adding reference to [PR](https://github.com/irina-mitrea-luxoft/gateway/pull/3).

irina-mitrea-luxoft commented:
@msalavastru

jitsni commented on transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/networkinterface/resolution/NetworkInterfaceResolutionIT.java:
```diff
@@ -0,0 +1,115 @@
+    @Rule
+    public TestRule chain = createRuleChain(gateway, robot);
+
+    @Specification("network.interface.accept")
```
can we use the scripts from specification.tcp ?

jitsni commented on transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/networkinterface/resolution/NetworkInterfaceResolutionIT.java:
```diff
@@ -0,0 +1,115 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.transport.wsn.networkinterface.resolution;
```
This is a wrong package.  I suppose we can add tests to transport/nio module.

I also think that the test should be like TcpAcceptorIT.java (it shouldn't use any proxy service etc). Then there may some test where gateway could be started using a config file with network syntax.

irina-mitrea-luxoft commented:
@jitsni Hello, Jitu. Added additional changes to [PR](https://github.com/irina-mitrea-luxoft/gateway/pull/3) related to defining infrastructure for using either java.net.URI, or NetworkInterfaceURI types.

jitsni committed:
Adding code to rewrite cookie domain, cookie path, location headers

jitsni commented on resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/Comparators.java:
```diff
@@ -24,6 +24,8 @@
 import java.util.Comparator;
 import java.util.List;
 
+import org.kaazing.gateway.resource.address.uri.URIUtils;
```
Is this needed ? There are quite few files in this PR like this.

irina-mitrea-luxoft replied:
Hello, Jitu, there are some static URIUtils methods used in this class. Therefore the import. The change in the file occurs due to a package refactoring.

jitsni commented on resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/ResourceFactories.java:
```diff
@@ -17,6 +17,8 @@
 
 import static java.lang.String.format;
 
+import org.kaazing.gateway.resource.address.uri.URIUtils;
```
Do you need the import ?

irina-mitrea-luxoft replied:
Hello, Jitu, there are some static URIUtils methods used in this class. Therefore the import. The change in the file occurs due to a package refactoring.

jitsni commented on resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/ResolutionUtils.java:
```diff
@@ -0,0 +1,172 @@
+    * @return
+    */
+   private static Enumeration<NetworkInterface> cloneInterfaces(List<NetworkInterface> interfaces) {
+       List<NetworkInterface> clone = new ArrayList<NetworkInterface>();
```
new ArrayList(interfaces) should do instead adding them separately in a loop

jitsni commented on resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/uri/URIAccessorFactory.java:
```diff
@@ -0,0 +1,41 @@
+ * Factory returning the appropriate UriAccessor
+ *
+ */
+public class URIAccessorFactory {
```
Seems like it is getting over-engineered with URIAccessorFactory, URIAccessor, GenericURI, URIWrapper ...

irina-mitrea-luxoft replied:
Hello, Jitu. I wanted to have two different implementations for the methods common for URI types which I wanted to have extracted in an interface (URIAccessor):
1. UriWrapper would be a wrapper over java.net.URI since java.net.URI is final (wanted to have a class which would implement URIAccessor)
2. NetworkInterfaceURI implements URIAccessor and provides NetworkInterface-specific parsing

GenericURI allows instantiation of the appropriate URI type using a factory (URIAccessorFactory) in order to reduce boilerplate code from the calling contexts.

irina-mitrea-luxoft commented:
@jitsni Hello, Jitu, I have added some more changes to my PR related to parsing of URIs.

Commit:
Added new test methods from k3po wse spec project to wseb connector and acceptor spec tests ClosingIT. All methods are now enabled and passing. Fixed WsebSession to report the session closed event when the session is implicitly closed due to abrupt close of transport rather than wsebSession.close() being called.

Commit:
Fixed rule chain ordering in wseb connector spec test BinaryIT so it doesn't fall over on expected IoException when k3po abruptly closes the upstream.

msalavastru committed:
Moved the Forwarded headers logic to HttpProxyServiceHandler

msalavastru committed:
Removed unused imports from HttpProxyServiceHandler

Commit:
Added first version of preferedIPV4 test.

msalavastru committed:
Changed the token for the 'Forwarded' header from 'server' to 'by'

irina-mitrea-luxoft committed:
Removed network interface resolution from the ResourceAddress RESOLVER

Commit:
fixed test

irina-mitrea-luxoft commented:
@jitsni Hello, Jitu, I have moved the network interface syntax resolution from ResourceAddress RESOLVER to TcpResourceAddressFactorySpi.

Commit:
Completed WsebConnector specification test ControlIT and made minor enhancement to ClosingIT to verify the type of exception.

Commit:
Added WsebConnector specification test CommandIT.

Commit:
Added wseb connector spec test DownstreamIT. Added ResponseFuture to HttpConnectSession so WsebConnector can validate downstream response headers immediately to get the test to pass. Cleaned up compilation warnings in HttpConnector (including removing dead code). Fixed failures in WsebConnectorLoggingIT by correcting the expected log messages.

Commit:
Fixed two failures in HttpProxyStreamingIT causes by my prior changes. Fix is to HttpConnector.ReadHandler.doSessionClosed: make sure we fire the session closed event when transport is (legimitately) closed after sending response with "Connection: closed" header.

irina-mitrea-luxoft committed:
Added minor refactoring for package-private access

irina-mitrea-luxoft committed:
Added fixes for supporting service resolution

irina-mitrea-luxoft committed:
Added minor test fix

irina-mitrea-luxoft committed:
Added fix for clustering

irina-mitrea-luxoft committed:
Added support for *.transport

irina-mitrea-luxoft committed:
Added support for *.bind

Commit:
Added wseb connector spec test OpeningIT. Cleaned up some compilation warnings in WsebAcceptor. Fixed create handshake validation in WsebConnector to check response content type and downstream and upstream URIs. Fixed WsebSession.cannotWrite to return true if there's no write address available (case of failed connector create handshake).

irina-mitrea-luxoft committed:
Fixed test

Commit:
Updated a comment in wseb connector spec test BinaryIT.

jitsni commented on resource.address/tcp/src/main/java/org/kaazing/gateway/resource/address/tcp/TcpResourceAddressFactorySpi.java:
```diff
@@ -198,7 +210,8 @@ else if (bindAddress instanceof String) {
 
-        URI uriOriginal = URI.create(original);
+        // TODO: find appropriate implementation
+        URI uriOriginal = URI.create(location);
```
As said earlier, we will use String for externalURI in ResourceAddress. So it will become

```
    public ResourceAddress(ResourceAddressFactorySpi factory, String externalURI, URI resourceURI) {
```

That way original can be passed as it is.

Commit:
Fixed WsebConnector to use /cbm instead of /cb (which is obsolete, only used by old clients). Fixed a bug in WsebSession.WsebSessionProcessor.cannotWrite. Changed some script names in the wse spec tests to adapt to changes made in k3po.

Commit:
Merge branch 'develop' of https://github.com/kaazing/gateway into wse_spec_tests

irina-mitrea-luxoft committed:
Added udp support (pending: tests supporting udp)

msalavastru committed:
Adjusted the forwarded headers to be comma separated and added a new config property 'use-forwarded' to enable/disable the use of forwaded headers in a http.proxy service

msalavastru committed:
Merge remote-tracking branch 'upstream/develop' into issue.547

# Conflicts:
#	service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyService.java

msalavastru committed:
Reverted the changes in the pom.xml for the http.proxy and removed unnecessary import in HttpProxyService

msalavastru committed:
Minor checkstyle fix

Commit:
Added fixes based on code review

irina-mitrea-luxoft committed:
Minor change

Commit:
Merge pull request #414 from vmaraloiu/fix_IndexOutOfBounds

Fixed issue regarding IndexOutOfBounds when no addresses available

Commit:
Added new wseb acceptor spec test ProxiesIT.shouldReceiveSecureRedirectFromServerOnProxyModeRequest(), ignored for now until tickets#321 gets fixed).

Commit:
Remove some dead code in WsebAcceptor (it's not needed for secure downstream redirect, that is done in WsebDownstreamHandler).

Commit:
Applied code review feedback.

msalavastru committed:
Added two k3po tests for the use of forwarded headers in http.proxy

Commit:
Fixed org.kaazing.gateway.server.test.Gateway testware class not to swallow exceptions caused by configuration issues, such as connect and accept options with names which are invalid XML element names (like "http[http/1.1].transport"). This was causing some tests to appear to succeed because they were not testing the functionality they thought they were.

jitsni committed:
Merge pull request #420 from cmebarrow/develop

Fixed org.kaazing.gateway.server.test.Gateway testware class not to swallow exceptions

Commit:
Merge branch 'develop' of https://github.com/kaazing/gateway into wse_spec_tests

Commit:
#421: added test case WsebRealignmentIT. Fixed the issue in DefaultHttpSession by changing readRequest member variable into a "deferred reads" queue.

irina-mitrea-luxoft committed:
Added changes for parametrized TcpAcceptorIT test considering network interface support

irina-mitrea-luxoft committed:
Fixed regular expression based on itest

Commit:
Added wseb acceptor spec test UpstreamIT.shouldAllowMultipleSequentialUpstreamRequests.

Commit:
Added wseb spec test UpstreamIT.shouldRejectZeroContentLengthUpstreamRequest. Fixed WsebUpstreamHandler to reject upstream request with zero content length. Fixed ITUtil.createRuleChain to terminate the chain with k3po.around(timeout) so test timeouts display the script differences instead of just reporting a timeout exception.

Commit:
Altered the logic in HttpProtocolCodecFilter.filterWrite not to commit the session (when http) if the child (httpxe) session is closing, in order to avoid defeating the logic that calculate content length when the session is closed, and because if child is closing the session will get closed anyway, which does the commit.

Commit:
Uncommented the lines for "http#[^wseb#]*wseb#[^ ]* [^/]*:\\d*] CLOSED" and the TODO items in WsebConnectorLoggingIT since these messages are now being correctly logged.

irina-mitrea-luxoft committed:
Added hazelcast-client and hazelcast-cloud as dependencies

Commit:
Change k3po dependency to latest release containing the wse spec test fixes.

Commit:
Remove an unwanted comment from pom.xml

jitsni committed:
Merge pull request #418 from cmebarrow/wse_spec_tests

Wse spec tests and changes for wse spec compliance

kaazing-build committed:
Merge branch 'release/5.0.1.46' into develop

kaazing-build committed:
Prepare release 5.0.1.46

kaazing-build committed:
Merge branch 'release/5.0.1.46'

kaazing-build committed:
Merge tag '5.0.1.46' into develop

Released 5.0.1.46

irina-mitrea-luxoft committed:
Added unit tests

Commit:
Moved DuplicateBindTest (transport.wsn -> transport.wse)

Commit:
Added changes for fixing test: use of ExpectedException rule to assert exception

Commit:
Moved DuplicateBindTest (*gateway/server/transport/wseb -> *gateway/transport/wseb)

Commit:
Cleaned up WsebDownstreamHandler and WsebUpstreamHandler by removing the unnecessary getWsebSession method and some log messages that are not needed because LoggingFilter now gives us enough diagnostics.

Commit:
(#424) Fixed WsebConnectorLoggingIT.shouldLogProtocolException to wait for the closed even to fire to make sure context.assertIsSatisfied does not race ahead of it.

Commit:
Fixed WsebConnectorLoggingIT to remove a workaround for close handshake not implemented (now that is fixed).

Commit:
Merge pull request #422 from irina-mitrea-luxoft/fix_HazelcastDependencyInBuild

Added hazelcast-client and hazelcast-cloud as dependencies

Commit:
(#424) Changed WsebDownstreamHandler.sessionOpened not to abruptly close the downstream if the wsebSession is closing. This allows processing to continue into WsebSession.attachWriter0 which deals with the isClosing case, and is enhanced to send the CLOSE frame if not already sent. This should fix failures in TextIT.clientSendTextInvalidUTF8 involving the downstream being closed with the wrong content type and no close frame.

irina-mitrea-luxoft committed:
Added parametrized TcpConnectorIT

Commit:
(#424) Changed WsebConnector.createHandler to include values from the supported protocols option on the resource address in the X-WebSocket-Protocol header, in order to be consistent with WsnConnector. Altered wseb connector specification test OpeningIT to use that option to set multiple values for the header. Fixed setting of the X-WebSocket-Protocol and X-WebSocket-Extension headers in WsebConnector create session initializer to use a separate header line for each value, because it typifies the code and is more easily compatibible with the wse specification tests. Fixed validation of X-WebSocket-Protocol and X-WebSocket-Extensions header in WsebConnector.readHandler.doSessionClosed, including allowing for extension parameters.

Commit:
Logging improvement: altered HttpConnector to call LoggingFilter.moveAfterCodec when connecting so the HTTP create request gets written legibly to the log file (instead of just the binary form).

Commit:
Updated pom.xml to pull in latest release k3po version with the needed wse spec test fixes.

Commit:
Undid two unnecessary changes to WsebConnector (we should continue to use setWriteHeader for single valued headers).

Commit:
Changed accept uri scheme to wse in DuplicateBindTest

Commit:
Changed MethodExecutionTrace to investigate Travis build failure "Killed" in transport.wsn.

Commit:
Changed MethodExecutionTrace to check for used memory (total - free) rather than just total.

Commit:
Catchup merge

Commit:
wseb connector spec test fixes: In OpeningIT, close the wsebSession in the success methods so the expectations will consistently be met (especially: to avoid exceptionCaught caused by early termination of IOSession due to k3po closing the TCPConnection). In DownstreamIT, waited for sessionClosed expectation to be met before allowing the test method to exit.

Commit:
Enhanced MethodExecutionTrace.healthCheck to attempt gc if memory usage is over 500MB.

Commit:
Updated the tests as per the feedback. Added ExtendedHandshakeBalancerIT.java in gateway/service/http/balancer to avoid circular dependency.

Commit:
Applied code review feedback to indicate a client only test.

irina-mitrea-luxoft committed:
Added tcp.bind and http.transport tests and fixed tcp.bind issue when no brackets provided

irina-mitrea-luxoft committed:
Added fix for udp.bind and tests for udp.bind and udp uri network interface syntax

irina-mitrea-luxoft committed:
Added transport URI refactored to String test fixes

Commit:
Minor formatting change

Commit:
wseb transport: fixed the logging in WsebSession to use the transport logger (passed in from WsebAcceptor or WsebConnector) instead of the logger called "session.scheduler" inherited from AbstractWsBridgeSession.

Commit:
wseb transport: fixed wseb connector spec test OpeningIT, shouldEstablishConnection.. methods to allow exceptionCaught(IOException) since that may happen depending on timing of context.assertIsSatisfied and k3po closing connections because there is no close handshake in the k3po scripts.

Commit:
Added comments to indicate client only tests

Commit:
Merge pull request #417 from sanjay-saxena/feature/httpx

HTTPX: Tests for Extended Handshake

Commit:
wseb transport: fixed WsebConnectorLoggingIT, shouldEstablishConnection.. methods to allow exceptionCaught(IOException) since that may happen depending on timing of context.assertIsSatisfied and k3po closing connections because there is no close handshake in the k3po scripts.

Commit:
(#424) Fixed WsebDownstreamHandler.reconnectSession to write the CLOSE and RECONNECT commands before closing the downstream if the wsebSession is closing. Fixed wse acceptor spec test ClosingIT.clientAbruptlyClosesUpstream to be less strict in checking the time taken for the wseb session to be closed. Also: removed some dead code from WsebSession (we now always align upstream and downstream).

Commit:
(#424) Fixed wseb acceptor spec test ClosingIT.shouldIgnoreDataFromServerAfterCloseFrame not to close the wsebSession since the server is initiating the close. Removed some unwanted code in WsebSession (CLOSE_WRITTEN state).

Commit:
(#423) Altered pom.xml to specify argLine parameter for failsafe plugin to limit heap size to 1024 MB. Removed the calls to System.gc from MethodExecutionTrace.healthCheck.

Commit:
Remove an unwanted Thread.sleep (added for testing purposes) from WsebSession.

Commit:
Removed unnecessary setup method from HttpProxyServiceTest (which was causing a log4j error message) and HttpProxySecureTest. Ignored wseb acceptor spec test ClosingIT.clientAbruptlyClosesUpstream pending resolution of #427.

Commit:
Applied review comments to ClosingIT.

Commit:
Merge pull request #426 from cmebarrow/issue_gateway_424

Changes to resolve build failures in wseb transport

Commit:
Fixed pom.xml in mina.netty to use the correct versions of the check style plug-in and code.quality, and correctly override the checkstyle plug-in configuration.

Commit:
(#427) Fixed the issue of exception stack traces "expected current thread ... to match..." happening when client abruptly closes the emulated WebSocket upstream or downstream by deferring the handling of the closed event in IoSessionChannelHandler is the session is not io registered (meaning it's in the process of being realigned). Reenabled ClosingIT.clientAbruptlyClosesUpstream.

Commit:
Merge branch 'develop' of https://github.com/kaazing/gateway into issue_427_alignment

Commit:
(#427) Further fix for the issue of exception stack traces "expected current thread ... to match...": in IoSessionChannelHandler.channelClosed, schedule the closed processing onto the new ioExecutor if current thread is still the old execution thread, which can happen because org.jboss.netty.channel.socket.nio.AbstractNioChannel.setWorker only schedules a task to unregister or register instead of doing it immediately. Also updated k3po version to pick up a fix in a test script.

Commit:
Merge pull request #428 from cmebarrow/issue_427_alignment

Issue 427: thread alignment violation in wse due to race with client close

irina-mitrea-luxoft committed:
Added fixes for clustering, added gateway config parser tests and parameterized tests in Tcp and Udp ResourceAddressFactorySpis

irina-mitrea-luxoft committed:
Added @ exclusion for matchin IPv6 in UdpResourceAddressFactorySpi and added ResolutionTestUtils for all component tests except NetworkInterfaceResolutionIT

irina-mitrea-luxoft committed:
Moved NetworkInterfaceResolutionIT to gateway.service.proxy

irina-mitrea-luxoft committed:
Added changes for cluster balancer service test with network interface

Commit:
Throw error on specific circumstances when PreferedIPv4 flag is true and the host IP is IPV6

irina-mitrea-luxoft committed:
Added error message when spaces in host name and no square brackets used

Commit:
Applied feedback.

msalavastru committed:
Changed the http proxy service properties for the use of forwarded headers to include 3 options: inject,ignore and exclude

irina-mitrea-luxoft committed:
Added test fixes after Linux integration and network interface syntax tests

irina-mitrea-luxoft committed:
Added minor changes to formatting

irina-mitrea-luxoft commented:
@jitsni Hello, Jitu. I have checked in the interface tests shared with you in the [excel](https://docs.google.com/spreadsheets/d/1glw5ZkIMLEe5quYceuPfOjGPrsGeAGrLAj1P0fE41W8/edit#gid=0). For subinterface tests I have added changes only within NetworkInterfaceURITest.

I am unsure as to how add more integration subinterface testing due to the fact that I would be unable to mock the resolution output. One way to do this would have been to use the ResolverOption on the ResourceAddress with its getAllByName method extended with my resolution (see ResourceAddress row 437; WildcardNameResolver is very similar to what would be needed except for the fact that it returns all the interfaces and subinterfaces resolved). Basically I could have mocked its resolution within TcpResourceAddressFactorySpiTest and UdpResourceAddressFactorySpiTest (similarly to how it is already done in these test classes).

However, I had previously performed this resolution here and at code review I was advised to move it away from here to TcpResourceAddressFactorySpi, respectively UdpResourceAddressFactorySpi. These are now located under
- TcpResourceAddressFactorySpi line 151
- UdpResourceAddressFactorySpi line 150

Should we move the resolution back to the ResolverOption? How should we proceed regarding this?

msalavastru committed:
Minor code refactoring for the use-forwarded methods

irina-mitrea-luxoft committed:
Added changes for supporting String original instead of URI in ResourceAddresses

irina-mitrea-luxoft commented:
@jitsni Hello, Jitu, apart from the [above inquiry](https://github.com/irina-mitrea-luxoft/gateway/pull/3#issuecomment-187738375) on subinterface testing, can you please give me your feedback on the current PR state? Apart from the above comment regarding testing, I haven't yet managed to perform your suggested changes of no longer using URIAccessorFactory and URI Wrapper classes. Thank you, Irina.

jitsni commented:
I think testing sub interfaces is not critical. Did you verify manually ?

BTW, some tests fail on mac during build since those resolvers(in TcpResourceAddressFactorySpiTest.java:168) need to deal with
"fe80:0:0:0:0:0:0:1%lo0" host names.

irina-mitrea-luxoft commented:
@jitsni Hello, Jitu.

> I think testing sub interfaces is not critical. Did you verify manually ?

If the existing testing should suffice I think the current approach should be fine. I will do some more manual testing on the topic

> BTW, some tests fail on mac during build since those resolvers(in TcpResourceAddressFactorySpiTest.java:168) need to deal with
> "fe80:0:0:0:0:0:0:1%lo0" host names.

I understand, I will adjust te tests to support more generic scenarios. I had tested TcpResourceAddressFactorySpiTest and UdpResourceAddressFactorySpiTest in Windows and Linux and the resolution seemed to follow a different pattern. I will adjust this.

msalavastru committed:
Updated the k3po script for forwarded headers where multiple chained proxies are used

Commit:
Applied feedback.

Commit:
Ported code review changes from 4.x.

irina-mitrea-luxoft committed:
Added fixes based on tests run on Linux

irina-mitrea-luxoft commented:
@jitsni @sbadugu Hello, Jitu. I have checked in the changes for fixing the issues which I had encountered on Linux subinterfaces. Any feedback provided on the pull request would be very useful in order to understand what changes would be needed. Thank you very much, Irina.

jitsni committed:
Removing ununsed imports

jitsni committed:
Merge remote-tracking branch 'upstream/develop' into httpxe.spec.tests

Conflicts:
	pom.xml

irina-mitrea-luxoft committed:
Minor fixes

jitsni committed:
* Validating http requests (http method, origin, headers)
* Added negative tests for origin security
* Added a httpxe spec config property to turn on the httpxe specification
  behaviour.

irina-mitrea-luxoft committed:
Added changes based on code review

jitsni committed:
Removing http proxy service properties from schema since it didn't
allow properties to be specified in any order. Now, the property
names are validated in the service initialization.

Commit:
Merge pull request #432 from jitsni/http.proxy.redirect.cookie

Adding code to rewrite cookie domain, cookie path, location headers

irina-mitrea-luxoft committed:
Added fix for UriUtils resolve method

irina-mitrea-luxoft committed:
Fix for connect options

irina-mitrea-luxoft committed:
Balancer fixes

irina-mitrea-luxoft committed:
Added merge from string_instead_of_uri branch

irina-mitrea-luxoft commented:
@jitsni Hello, Jitu. As mentioned in today's meeting, some code review feedback would be very useful in order to understand which changes to apply for the current PR. Since the [refactoring of URIs to String](https://github.com/kaazing/gateway/pull/395) would be merged together with this one some feedback would be welcomed also on the respective PR.

Also: should we throw an exception when the network interface syntax is provided in the connects? Initially it was discussed that we should not support network interface syntax for connects, but our current approach supports it.

Thank you,
Irina.

jitsni commented:
I think it would be ok to have connects with network interface syntax.

I think it would be good to have corresponding PR for enterprise.gateway. Then we can merge them one after another. Otherwise other enterprise.gateway PRs would be blocked.

irina-mitrea-luxoft commented:
@jitsni @sbadugu Hello, Jitu. Thank you for your feedback, can you please let me know what your feedback is on the current implementation? Any code review comments would be very useful.

Thank you,
Irina.

irina-mitrea-luxoft committed:
Minor fixes for gateway.management and transports

Anisotrop committed:
Merge remote-tracking branch 'parallel/duplicate_bind_test_fix' into duplicate_bind_test_fix

Import changes form ACojanu's fork.

msalavastru committed:
Merge remote-tracking branch 'upstream/develop' into issue.547

# Conflicts:
#	service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyService.java
#	service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

msalavastru committed:
Implemented code review changes

jitsni committed:
Using one method (instead of one for wrapped and the other for unwrapped)
to write responses

jitsni commented:
- ResolutionTestUtils can be moved to base package (for e.g o.k.g.resource.address.tcp) in a module. No need to have such extra package names.
- NetworkInterfaceSyntaxException can be removed (moreover its constructor throws IAE, that means it won't be of any use.)
- URIAccessor can be removed
- NetworkInterfaceURI can be static inner class of URIUtils
- ResolutionUtils#getAllByName: you can eliminate one of the List local variables
- javadoc - you have empty param, return. If you don't plan to add any javadoc, just remove it.
- ResolutionUtils.resolveDeviceAddress:112 - IDE says subInterfaces != null is always true, so you can remove the condition.
- I am ok if you keep the existing methods on AbstractServiceConfigurationBuilder.java and test classes and deprecate them (impl will delegate to methods that take String parameter). You can add new methods with String parameter. May be that will reduce the number of files (IT.java), but it is up to you. But this way every test is forced to move and in a way it is good
-

Anisotrop committed:
Simplified DuplicateBindTest to use GatewayRule and ExpectedException
rule.

Commit:
Ported review changes.

Commit:
Minor checkstyle fixes.

msalavastru committed:
Updated the k3po scripts in gateway.service.http.proxy to be compliant with the use of forwarded headers being disabled (ignored) by default

irina-mitrea-luxoft committed:
Applied code review changes

irina-mitrea-luxoft committed:
Applied code review changes in regard to using inner class for NetworkInterfaceURI

irina-mitrea-luxoft commented:
@jitsni @sbadugu Hello, Jitu, thank you for your feedback, I have applied the changes as per your [comment](@sbadugu) above. Regarding the point 
>
I am ok if you keep the existing methods on AbstractServiceConfigurationBuilder.java and test classes and deprecate them (impl will delegate to methods that take String parameter). You can add new methods with String parameter. May be that will reduce the number of files (IT.java), but it is up to you. But this way every test is forced to move and in a way it is good

I had switched from URI to String to all test files as per our [initial discussion](https://github.com/kaazing/gateway/pull/372#discussion_r48917262). We can have similar method with different prototypes, but that means modifying a significant part of this PR. Also, as you state above, this means forcing the usage of the new String objects. Please let me know what your opinion is on this.

irina-mitrea-luxoft commented:
@jitsni @sbadugu Hello, Jitu, thank you for your feedback, I have applied the changes as per your [code review comments](https://github.com/irina-mitrea-luxoft/gateway/pull/3#issuecomment-191036637) above. Please provide me your feedback. Regarding the point 

> I am ok if you keep the existing methods on AbstractServiceConfigurationBuilder.java and test classes and deprecate them (impl will delegate to methods that take String parameter). You can add new methods with String parameter. May be that will reduce the number of files (IT.java), but it is up to you. But this way every test is forced to move and in a way it is good

I had switched from URI to String to all test files as per our [initial discussion](https://github.com/kaazing/gateway/pull/372#discussion_r48917262). We can have similar methods with different prototypes (different parameters), but that would mean modifying a significant part of this PR. Also, as per the statement above, the current approach means forcing the usage of the new String objects in the new configurations. Please let me know what your opinion is on this.

jitsni committed:
Merge pull request #431 from vmaraloiu/fix_IndexOutOfBounds

Throw error on specific circumstances when PreferedIPv4 flag is true and the host IP is IPV6

jitsni committed:
Merge pull request #434 from Anisotrop/duplicate_bind_test_fix

Duplicate bind test fix

jitsni commented:
Let us keep your changes as you have already made them.

irina-mitrea-luxoft committed:
Added minor changes at integration with enterprise.gateway

jitsni committed:
Incorporating some more origin tests

Commit:
Made changes to include the gatewayContext to the startingGateway and stoppingGateway methods from GatewayObserverApi

Commit:
Minor correction of the IllegalArgumentException message throwen from
checkForTralingSlashes()

Commit:
Merge pull request #435 from ilyaanisimov-kaazing/slash_message

Minor correction of the IllegalArgumentException message throwen from checkForTralingSlashes()

irina-mitrea-luxoft committed:
Added minor fixes

irina-mitrea-luxoft committed:
Added minor test fixes

Anisotrop committed:
fiex line ending on /DuplicateBindTest.java

irina-mitrea-luxoft commented:
Adding reference to [PR](https://github.com/kaazing/gateway/pull/437) holding changes for both URI to String refactoring and network interface support.

jitsni committed:
Merge pull request #436 from Anisotrop/fix_crlf_DuplicateBindTest

Fix line endings on DuplicateBindTest.java

jitsni committed:
Proxy service fails the connect future when the corresponding accept
session is closed

jitsni committed:
Merge remote-tracking branch 'upstream/develop' into socks.over.wseb

Commit:
Merge pull request #413 from jitsni/httpxe.spec.tests

Httpxe.spec.tests

Commit:
Merge pull request #439 from jitsni/socks.clear.requests

Proxy service fails the connect future when the corresponding accept is closed

kaazing-build committed:
Prepare release 5.0.1.47

kaazing-build committed:
Merge branch 'release/5.0.1.47' into develop

kaazing-build committed:
Merge branch 'release/5.0.1.47'

kaazing-build committed:
Merge tag '5.0.1.47' into develop

Released 5.0.1.47

Anisotrop committed:
Improved parsing of IPv6 address and added unit test.

Commit:
Removed gatewayContext from the newInstance method in GatewayObserver

Commit:
Delete accidentally added file transport/wseb/dumpfile.pcap.

Commit:
Merge pull request #441 from cmebarrow/delete_dumpfile

Delete accidentally added file transport/wseb/dumpfile.pcap.

Commit:
Merge pull request #440 from mgherghe/license

Added gatewayContext to startingGateway and stoppingGateway methods from GatewayObserverApi

jitsni committed:
* reader is thread-aligned
* X_WEBSOCKET_PROTOCOLS, X_WEBSOCKET_EXTENSIONS contain null values, so
guarding aganist them

Anisotrop committed:
Modify unit test to match the external URI.

NicoletaOita committed:
Added test for web socket compliance (maximum message size and maxmimum
lifetime).

Anisotrop committed:
Fix line endings on test file.

Anisotrop committed:
fix line endings for impl file

irina-mitrea-luxoft committed:
Added changes for checking http.proxy loops

msalavastru committed:
Removed the code for the 'Forwarded' header from HttpSubjectSecurityFilter

Commit:
Merge pull request #392 from jitsni/socks.over.wseb

Wseb session thread migration

irina-mitrea-luxoft committed:
Added changes from upstream and fixed conflicts; fixed build

Anisotrop committed:
Implement a new close message type triggerd by a different exception.

Commit:
resolves kaazing/tickets#111

irina-mitrea-luxoft committed:
Minor first set of changes at code review

irina-mitrea-luxoft committed:
Added changes based on code review; service id in via header stored as uuid

irina-mitrea-luxoft committed:
Added logic minor refactoring to use streams

irina-mitrea-luxoft committed:
Terminated proxy connection prior to connecting when loop detected

irina-mitrea-luxoft committed:
Moved NetworkInterfaceResolutionIT to upper package

Commit:
Fixed pipe error message.

jitsni committed:
Merge pull request #437 from irina-mitrea-luxoft/network_interface_support

Pull request holding URI to String and network interface support changes

Commit:
Altered log4j-config.xml in distribution (default OOTB log4j config) to add logger "transport.tcp" since it is often useful for auditing or support purposes. Altered distribution\src\main\assembly\filter.properties to change "KAAZING Gateway" to "Kaazing Gateway" in line with marketing directives.

kaazing-build committed:
Merge branch 'release/5.0.1.48' into develop

kaazing-build committed:
Prepare release 5.0.1.48

kaazing-build committed:
Merge branch 'release/5.0.1.48'

kaazing-build committed:
Merge tag '5.0.1.48' into develop

Released 5.0.1.48

Commit:
Applied feedback.

Commit:
Merge remote-tracking branch 'upstream/develop' into fix_pipe_error

Commit:
Ported changes for Cache-control support feature (#560)

irina-mitrea-luxoft committed:
Addede changes as per code review

jitsni committed:
Merge pull request #448 from vmaraloiu/fix_pipe_error

Fixed pipe error message.

irina-mitrea-luxoft committed:
Fix to loop detection validation

irina-mitrea-luxoft committed:
Adjusted *.rpt scripts with tcp transport Via headers in order to support new <1.1 kaazing-uuid> format

irina-mitrea-luxoft committed:
Merge remote-tracking branch 'upstream/develop' into fix_http.proxy_loop

Conflicts:
	service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

irina-mitrea-luxoft committed:
Added fixes after merging changes from upstream

irina-mitrea-luxoft committed:
Fixed remaining *.rpt scripts with Via header

NicoletaOita committed:
Modified the tests according to review.

NicoletaOita committed:
Merge remote-tracking branch 'upstream/develop' into ws_spec_compliance_tests

msalavastru committed:
Merge remote-tracking branch 'upstream/develop' into issue.547

# Conflicts:
#	service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

NicoletaOita committed:
Changed the service accept from URI to String.

Commit:
Added fixes for existing ITs

msalavastru committed:
Minor fixes due to URI to String refactoring

Commit:
Merge remote-tracking branch 'upstream/develop' into cache-control

Commit:
Fixed compilation errors after merging changes from upstream

Commit:
Update request1.request2.response1.response2.rpt

jitsni committed:
Merge pull request #446 from dpwspoon/roadmap111

fixes issue where k3po regex might match on tcp packet fragmentation

jitsni committed:
Merge pull request #443 from irina-mitrea-luxoft/fix_http.proxy_loop

Connection closed when http.proxy loops encountered

jitsni committed:
Fixing NPE when an exception is caught but wsn session is not yet
created.

Commit:
Merge pull request #452 from jitsni/wsn.connector.npe

Fixing NPE when an exception is caught but wsn session is not yet created

Commit:
Updated GatewayObserverSpi to provide init hook on gateway that gives access to base gateway properties, and an ability to access the injectable properties

mgherghe committed:
Merge pull request #2 from dpwspoon/mgherghe_check2

Mgherghe check2

msalavastru committed:
Merge remote-tracking branch 'upstream/develop' into issue.547

# Conflicts:
#	service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

msalavastru committed:
Resolved merging conflicts

jitsni committed:
Merge pull request #415 from msalavastru/issue.547

Support proxying to application servers that generate dynamic resource addresses

NicoletaOita committed:
Renamed a method after review.

kaazing-build committed:
Merge branch 'release/5.0.1.49' into develop

kaazing-build committed:
Prepare release 5.0.1.49

kaazing-build committed:
Merge branch 'release/5.0.1.49'

kaazing-build committed:
Merge tag '5.0.1.49' into develop

Released 5.0.1.49

Commit:
Enable the ignored test in WsxAcceptorLoggingIT.

jitsni committed:
Merge pull request #453 from cmebarrow/forward_port_gateway_server_111

Enable the ignored test in WsxAcceptorLoggingIT.

Commit:
Use project version as double layer of indirection would cause update version plugin to fail, and sometimes allow mina.netty to build prior to mina being built, it the same version was already built

NicoletaOita committed:
Updated the k3po version to the last one.

Anisotrop committed:
Merged with latest develop from upstream

Commit:
Merge pull request #449 from cmebarrow/add_tcp_logger_to_ootb_config

Add transport.tcp logger to the OOTB log4j config.xml file

Commit:
Minor refactoring

jitsni committed:
Merge pull request #445 from Anisotrop/fix_error_code_for_payload_too_long

Gateway returns incorrect error code when payload is too long (1002 instead of 1009)

Commit:
transport/ssl: Accomodate change of transport option from type URI to string in VirtualHostKeySelectorTest.

jitsni committed:
Merge pull request #458 from cmebarrow/issue_457

VirtualHostKeySelectorTest: Accomodate change of transport option from type URI to String

Commit:
Merge branch 'license' of https://github.com/mgherghe/gateway into mgherghe-license3

Commit:
Changed instance names from listener to observer to be consistent with class names

jitsni committed:
Merge pull request #460 from dpwspoon/mgherghe-license3

nit fixes + Added initingGateway method in GatewayObserverApi

kaazing-build committed:
Merge branch 'release/5.0.1.50' into develop

kaazing-build committed:
Prepare release 5.0.1.50

kaazing-build committed:
Merge branch 'release/5.0.1.50'

kaazing-build committed:
Merge tag '5.0.1.50' into develop

Released 5.0.1.50

ahousing committed:
Merge pull request #454 from dpwspoon/build.version.diff

Use project version as double layer of indirection would cause update…

Anisotrop committed:
Refactoring taking into account code review and network interfaces syntax.

Anisotrop committed:
Improved unit tests.

Anisotrop committed:
Reverted regexp to match the rest of the network interfaces regexps.

Commit:
Removed double creation of TcpExtensionFactory

Commit:
Cleaned up code

Commit:
Merge pull request #447 from NicoletaOita/ws_spec_compliance_tests

Added test for web socket compliance (maximum message size and maxmimum lifetime).

irina-mitrea-luxoft committed:
Added changes for having the network interface syntax regexp exposed within a specific constant

kaazing-build committed:
Merge branch 'release/5.0.1.51' into develop

kaazing-build committed:
Prepare release 5.0.1.51

jitsni committed:
Merge pull request #462 from dpwspoon/develop

Removed double creation of TcpExtensionFactory

kaazing-build committed:
Merge branch 'release/5.0.1.51'

kaazing-build committed:
Merge tag '5.0.1.51' into develop

Released 5.0.1.51

Commit:
Merge pull request #451 from mgherghe/cache-control

Forward feature #560 from 4.x to 5.x: Implement the directives for Cache-Control header in Directory Service

jitsni committed:
Merge pull request #455 from Anisotrop/Anisotrop-fix_ipv6_accept_options

Anisotrop fix ipv6 accept options

irina-mitrea-luxoft committed:
Merge remote-tracking branch 'upstream/develop' into network_interface_support

Conflicts:
	resource.address/tcp/src/main/java/org/kaazing/gateway/resource/address/tcp/TcpResourceAddressFactorySpi.java

irina-mitrea-luxoft committed:
Added changes for conflict resolution with upstream

irina-mitrea-luxoft committed:
Added minor fixes and ResolutionUtilsTest->parseBindAddress call in UdpResourceAddressFactorySpi

irina-mitrea-luxoft commented:
Closing as this PR was created only for facilitating viewing diff.
